include/dlaf/eigensolver/band_to_tridiag/mc.h

```diff
@@ -248,11 +248,46 @@ struct BandBlock {
   }
 
 private:
-#ifdef DLAF_WITH_CUDA
+#ifdef DLAF_WITH_GPU
   bool isAccessibleFromGPU() const {
     cudaPointerAttributes attrs;
     DLAF_CUDA_CHECK_ERROR(cudaPointerGetAttributes(&attrs, mem_()));
+#ifdef DLAF_WITH_CUDA
     return cudaMemoryTypeUnregistered != attrs.type;
+#elif defined DLAF_WITH_HIP
+    // HIP 5 defines the following:
+    //
+    // /**
+    //  * Memory type (for pointer attributes)
+    //  */
+    // typedef enum hipMemoryType {
+    //     hipMemoryTypeHost,    ///< Memory is physically located on host
+    //     hipMemoryTypeDevice,  ///< Memory is physically located on device. (see deviceId for specific
+    //                           ///< device)
+    //     hipMemoryTypeArray,  ///< Array memory, physically located on device. (see deviceId for specific
+    //                          ///< device)
+    //     hipMemoryTypeUnified  ///< Not used currently
+    // } hipMemoryType;
+    //
+    // /**
+    //  * Pointer attributes
+    //  */
+    // typedef struct hipPointerAttribute_t {
+    //     enum hipMemoryType memoryType;
+    //     int device;
+    //     void* devicePointer;
+    //     void* hostPointer;
+    //     int isManaged;
+    //     unsigned allocationFlags; /* flags specified when memory was allocated*/
+    //     /* peers? */
+    // } hipPointerAttribute_t;
+    //
+    // TODO: What's the right thing to do here?
+    // Not useful documentation:
+    // - https://github.com/RadeonOpenCompute/ROCm_Documentation/blob/master/Programming_Guides/HIP-porting-guide.rst#cu-pointer-attribute-memory-type
```

Comment:
Fixed link: https://github.com/RadeonOpenCompute/ROCm_Documentation/blob/master/Programming_Guides/HIP-porting-guide.rst#CU-Pointer-Attribute-Memory-Type.

Reply:
I've removed most of the comment here. As far as I understand we don't really have a proper way of checking this in HIP, but we can always update it if we find a way in the future.